### ISSUE TITLE
Fix false positives for shared-line comments in nested blocks in rule-empty-line-before

### ISSUE THREAD

**Issue by gasman** [stylelint/stylelint#2919]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where a comment at the top of a nested block using the `rule-empty-line-before` rule returns false results. This appears to be a regression from 8.0.0, possibly triggered by #2827.

> Which rule, if any, is this issue related to?

`rule-empty-line-before`

> What CSS is needed to reproduce this issue?

```css
@media screen { /* comment */
    .foo {
        display: none;
    }
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "rule-empty-line-before": [ "always", {
      "except": ["after-single-line-comment", "first-nested"]
    }]
  }
}
```

> Which version of stylelint are you using?

`8.1.1` (also reproduced on current master https://github.com/stylelint/stylelint/commit/c856d321db2fd33bbef0c98b443a294877abe3ba)

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint test.css`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
test.css
 2:5  ✖  Expected empty line before rule   rule-empty-line-before
```

**Comment by jeddy3** [comment]
@gasman Thanks for the report and for using the template.

I can confirm this bug. It appears to be linked to the interplay of the two options as just `"except": ["first-nested"]` doesn't cause a violation.

Labelling as bug and help wanted. @nwoltman Could this be a regression triggered by #2827?

**Comment by nwoltman** [comment]
@jeddy3 This isn't exactly a regression, but it is a bug that was revealed by a combination of #2827 and @gasman's particular configuration. I just opened a PR that should fix this bug :)
